### PR TITLE
feat(axios): Add route pre hook that provides axios client to request

### PR DESCRIPTION
**Pull request by clok**
- Adds `axiosFactory` method with interceptors for errors that add helpful custom data that we use within our services.
- Adds an `axios` instance to the request plugin space. Thank you @mrlannigan 
- Updated README

## axios Client Route Usage

Passing `axios` plugin or route configuration will create Axios instances on the `request`. Use these instances to have the trace headers injected. This will allow for chained request tracing.

```js
await server.register({
    plugin: require('@goodwaygroup/lib-hapi-good-tracer'),
    options: {
        axios: {
            main: {}
        }
    }
});

const routes = [{
    method: 'GET',
    path: '/proxy/google',
    config: {
        tags: ['proxy'],
    },
    handler: async (request) => {
        const { axios } = request.plugins.goodTracer;
        return axios.main.get('https://google.com')
    }
}];

exports.routes = server => server.route(routes);
```

And with Route level configuration

```js
const routes = [{
    method: 'GET',
    path: '/proxy/google',
    config: {
        tags: ['proxy'],
        plugins: {
            goodTracer: {
                axios: {
                    inroute: {}
                }
            }
        }
    },
    handler: async (request) => {
        const { axios } = request.plugins.goodTracer;
        return axios.inroute.get('https://google.com')
    }
}];

exports.routes = server => server.route(routes);
```

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -155,4 +163,22 @@ exports.register = (server, options) => {
     }
 };
 
-exports.name = NAME;
+exports.addAxiosRoutePreHook = {
+    assign: 'axios',
```
Comment (bnormington) [inline lib/index.js:167]: Should this prehook assign name 'axios' be configurable?

File: lib/index.js [hunk 2]
```diff
@@ -155,4 +163,22 @@ exports.register = (server, options) => {
...
+    method: (request) => {
+        const factory = get(request, ['server', 'plugins', NAME, 'axiosFactory']);
+        const { traceUUIDHeader, traceDepthHeader, config } = get(request, ['server', 'plugins', NAME, 'axiosConfig'], {});
+
+        // provide axios client to request
+        const axiosConfig = Hoek.applyToDefaults(config, {
+            headers: {
+                common: {
+                    [traceUUIDHeader]: get(request, ['plugins', NAME, traceUUIDHeader]),
+                    [traceDepthHeader]: get(request, ['plugins', NAME, traceDepthHeader])
```
Comment (bnormington) [inline lib/index.js:177]: Who is responsible for incrementing the depthHeader?  I was expecting it to be incremented here so all new calls using axios would be using the new depth.

Comment (clok) [reply lib/index.js:177]: This library handles the incrementing internally.

The `incrementDepth` method is implemented here: https://github.com/GoodwayGroup/lib-hapi-good-tracer/blob/master/lib/index.js#L26

It is called in the `onRequest` event that is hooked into here: https://github.com/GoodwayGroup/lib-hapi-good-tracer/blob/master/lib/index.js#L106-L109

File: lib/index.js (hunk 1 above)
Comment (clok) [reply lib/index.js:167]: We could. I had considered it, but wasn't sure if we should. I am fine either way.

File: lib/index.js (hunk 2 above)
Comment (clok) [reply lib/index.js:177]: That is to say this library implements the features according to the spec laid out [in coding-standard](https://github.com/GoodwayGroup/coding-standards/blob/master/service-request-tracing.md)

File: lib/index.js (hunk 1 above)
Comment (clok) [reply lib/index.js]: @bnormington Just pushed a refactor to support this concept.

**Comment by clok** [comment]
Fixes #6

**Pull request merged**